src/utilities/include/utilities/Interval.h

```diff
@@ -33,14 +40,114 @@ class Interval
 	/**
 	 * @brief Construct a new Interval object from bounds and bound types
 	 *
-	 * @param lb
-	 * @param lbType
-	 * @param up
-	 * @param upType
+	 * @param lb The value of the lower bound (ignored if lbType == INFTY)
+	 * @param lbType The type of the lower bound
+	 * @param ub The value of the higher bound (ignored if ubType == INFTY)
+	 * @param ubType The type of the higher bound
+	 */
+	Interval(N lb, BoundType lbType, N ub, BoundType ubType)
+	: lower_(lb), upper_(ub), lowerBoundType_(lbType), upperBoundType_(ubType)
+	{
+		if (lbType == BoundType::INFTY) {
+			lower_ = std::numeric_limits<N>::min();
+		}
+		if (ubType == BoundType::INFTY) {
+			upper_ = std::numeric_limits<N>::max();
+		}
+	}
+
+	/// Check if the first interval is lexicographically smaller than the second.
+	/** Compare two intervals lexicographically. Note that this is not a partial
+	 * order of intervals in the mathematical sense, e.g., if [a, b] < [c, d],
+	 * you cannot infer that b < c. Instead, it is a total order used to put
+	 * Intervals into containers.
+	 * @param first The first interval to compare
+	 * @param second The second interval to compare
+	 * @return true if first is smaller than second
+	 */
+	friend bool
+	operator<(const Interval &first, const Interval &second)
+	{
+		auto to_tuple = [](const Interval &interval) {
+			return std::tie(interval.lower_,
+			                interval.lowerBoundType_,
+			                interval.upper_,
+			                interval.upperBoundType_);
+		};
+		return to_tuple(first) < to_tuple(second);
```

Comment:
Is that using a built-in tuple comparator or where is the operator located? Just wanted to make sure that the claim about ignoring INFTY-bounds is true.

Reply:
Yes, that's the built-in tuple comparator. `std::tie` creates a tuple that can then be compared using the usual tuple comparator. This avoids writing a custom comparator.

Reply:
Okay, but then is the value of a bound really ignored in case the bound is INFTY?

Reply:
Re INFTY bounds: As we enforce the value to be always the same in case that we INFTY bounds, this should be the case here. Should I add an assertion?